lib/rules/prefer-regex-literals.js

```diff
@@ -293,6 +296,51 @@ module.exports = {
             }
         }
 
+        /**
+         * Checks whether the given regex and flags are valid for the ecma version or not.
+         * @param {string} pattern The regex pattern to check.
+         * @param {string | undefined} flags The regex flags to check.
+         * @returns {boolean} True if the given regex pattern and flags are valid for the ecma version.
+         */
+        function isValidRegexForEcmaVersion(pattern, flags) {
+            const RegExpValidatorInstance = new RegExpValidator({ ecmaVersion: regexppEcmaVersion });
```

Comment:
Instances should not begin with an uppercase letter. We can probably also shorten to `validator` in this context:

```suggestion
            const validator = new RegExpValidator({ ecmaVersion: regexppEcmaVersion });
```